### PR TITLE
Update python client to API v0.11.1

### PR DESCRIPTION
**Pull request by sreev**
Update python client to API v0.11.1

### CODE REVIEW

File: marquez_client/client.py [hunk 1]
```diff
@@ -29,10 +29,15 @@
 _USER_AGENT = f'marquez-python/{VERSION}'
 _HEADERS = {'User-Agent': _USER_AGENT}
 
+_NAMESPACE_LENGTH_LIMIT = 1024
```
Comment (wslulciuc) [inline marquez_client/client.py]: _Minor:_ All names will be <= `1024`, so maybe define a single constant?

```python
_NAME = 1024
.
.
if len(namespace_name) > _NAME:
  raise ValueError('namespace_name length is {0}, must be <= {1}',
    len(namespace_name), _NAME)

# check job name using _NAME
# etc
```

File: marquez_client/client.py [hunk 2]
```diff
@@ -41,15 +46,24 @@ def __init__(self, host=None, port=None,
         self._namespace_name = namespace_name or os.environ.get(
             'MARQUEZ_NAMESPACE', DEFAULT_NAMESPACE_NAME
         )
-        self._api_base = f'http://{host}:{port}/{_API_PATH}'
+        self._api_base = f'{protocol}://{host}:{port}/{_API_PATH}'
+
+        if not port or port == 8080 or port == 80:
```
Comment (wslulciuc) [inline marquez_client/client.py]: Do we need this check? We set the port [here](https://github.com/MarquezProject/marquez-python/blob/master/marquez_client/client.py#L37) and default to port `8080` if not provided

File: marquez_client/client.py [hunk 3]
```diff
@@ -29,10 +29,15 @@
...
 _HEADERS = {'User-Agent': _USER_AGENT}
 
+_NAMESPACE_LENGTH_LIMIT = 1024
+_SOURCE_LENGTH_LIMIT = 1024
+_DATASET_LENGTH_LIMIT = 1024
+_JOB_LENGTH_LIMIT = 1024
 
 class MarquezClient(object):
-    def __init__(self, host=None, port=None,
+    def __init__(self, protocol=None, host=None, port=None,
```
Comment (wslulciuc) [inline marquez_client/client.py]: Nice! Adding an option for setting the `protocol` is a great idea 💯

Comment (wslulciuc) [reply marquez_client/client.py]: _Minor:_ The client would only support `https` and `http`. We can rename `protocol` to `ssl` and define it as a flag:

```python
class MarquezClient(object):
  def __init__(self, ssl=False, host=None, port=None, ...
```

And depending on the value of `ssl`, we can add `https` or `http` with the appropriate ports

File: marquez_client/client.py (hunk 1 above)
Comment (julienledem) [reply marquez_client/client.py]: Isn't NAME a bit misleading? _NAME_LENGTH_LIMIT is clearer what it is.

File: marquez_client/client.py [hunk 4]
```diff
@@ -80,16 +97,204 @@ def list_namespaces(self, limit=None, offset=None):
...
+
+        return self._put(self._url('/sources/{0}', source_name),
+                         payload=payload)
+
+    def get_source(self, source_name):
+        if not source_name:
+            raise ValueError('source_name must not be None')
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
```
Comment (julienledem) [inline marquez_client/client.py]: Great error messages. That improves user experience.
Possibly factor out the length check logic into a reusable function.

File: marquez_client/client.py [hunk 5]
```diff
@@ -80,16 +97,204 @@ def list_namespaces(self, limit=None, offset=None):
...
+            raise ValueError('STREAM type datasets must have schema_location')
+
+        if not physical_name:
+            raise ValueError('physical_name must not be None')
+
+        if not source_name:
+            raise ValueError('source_name must not be None')
+        if len(source_name) > _SOURCE_LENGTH_LIMIT:
+            raise ValueError('source_name length is {0}, must be <= {1}',
+                             len(source_name), _SOURCE_LENGTH_LIMIT)
```
Comment (julienledem) [inline marquez_client/client.py]: This is a good example that could be made more compact by reusing a function.
```
checkLength('namespace_name', namespace_name)
...
checkPresent('physical_name', physical_name)
checkLength('source_name', source_name)
```

File: marquez_client/client.py (hunk 2 above)
Comment (julienledem) [reply marquez_client/client.py]: Also, if the goal it to make the url look cleaner when using the default port, it looks like this is not correct if port is 8080. We can only omit the port if it's 80

**Review by julienledem — CHANGES_REQUESTED** [review]
This looks great overall.
Please review the logic around port 8080 (see bellow) and Willy's comments

File: marquez_client/client.py (hunk 1 above)
Comment (wslulciuc) [reply marquez_client/client.py]: Yep, good point! What I _really_ wanted to type was `_MAX_NAME_LENGTH`.

File: marquez_client/client.py (hunk 5 above)
Comment (wslulciuc) [reply marquez_client/client.py]: +1 for these util methods!

Comment (wslulciuc) [reply marquez_client/client.py]: Add maybe to be clear on the type of length check, we can define the method: `checkMaxLength(str, max_int)`

File: marquez_client/client.py (hunk 1 above)
Comment (sreev) [reply marquez_client/client.py]: done.
`_check_name_length`

File: marquez_client/client.py (hunk 3 above)
Comment (sreev) [reply marquez_client/client.py]: done.

File: marquez_client/client.py (hunk 5 above)
Comment (sreev) [reply marquez_client/client.py]: done.
multiple utility methods added.

File: marquez_client/client.py (hunk 4 above)
Comment (sreev) [reply marquez_client/client.py]: done.

File: marquez_client/client.py (hunk 2 above)
Comment (sreev) [reply marquez_client/client.py]: removed the check for 8080.

**Comment by sreev** [comment]
Thank you for the valuable comments, @julienledem @wslulciuc . python client looks much stable.

**Review by julienledem — APPROVED** [review]
LGTM!
Thank you for your contribution! 🥇

**Review by wslulciuc — APPROVED** [review]
@sreev Thanks for all these great changes! 💯 🥇

**Comment by wslulciuc** [comment]
Odd, I'll have to look into why CI for PRs are not running. I will merge and fix any issues with the `master` branch

**Pull request merged**